### PR TITLE
Allows user to delete External Cluster from the Cloud Provider

### LINKED ISSUE #10299: Delete Cloud Provider External Cluster
### Description of the feature you would like to add / User story

As a developer
I would like to delete eks/gke/aks clusters
in order to kkp external cluster mgmt

### Solution details

- Requirement 1
- Requirement 2
- Requirement 3

### PR DESCRIPTION
**Pull request by imharshita**
Signed-off-by: Harshita sharma <harshita.sharma6174@gmail.com>

**What does this PR do / Why do we need it**:

**Does this PR close any issues?**:
Fixes #10299 

**Special notes for your reviewer**:
"Action" header in req is used to decide if user wants to disconnect the external cluster or to delete a cluster from cloud provider.

Please suggest other name if this seems ambigious

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
Allows user to delete External Cluster from the Cloud Provider
```

### CODE REVIEW

**Comment by imharshita** [comment]
Cc @KhizerRehan  @ahmadhamzh

File: pkg/handler/v2/external_cluster/external_cluster.go [hunk 1]
```diff
@@ -303,6 +345,11 @@ type deleteClusterReq struct {
 	// in: path
 	// required: true
 	ClusterID string `json:"cluster_id"`
+	// The action is used to check if to delete a externalcluster or a actual cluster
+	// in: header
+	// name: Action
+	// Possible values: DeleteExternalCluster, DeleteProviderCluster
+	Action string `json:"action"`
```
Comment (lsviben) [inline pkg/handler/v2/external_cluster/external_cluster.go]: Maybe add here that by default, the behavior is to just delete the External cluster object.

Also not sure if  the difference for DeleteExternalCluster and DeleteProviderCluster are clear enough. Maybe the Disconnect/Delete that you use in code is less ambigous

**Pull request merged**